### ISSUE TITLE
Organize Token Gate Huddle calls from the event app

### ISSUE THREAD

**Issue by julien51** [unlock-protocol/unlock#13296]
When creating an event, it would be useful for users who want to create video calls to be able to select huddle and have the config be done from there automatically!

Here are the huddle docs: https://docs.huddle01.com/docs/apis/create-room/token-gated-room

cc @ramitphi

**Comment by julien51** [comment]
This could be really useful for community calls (including Unlock's or Celo's)

**Comment by Ramitphi** [comment]
thanks for creating the issue @julien51, will share it in our dev community.
just wanted to know if we have some designs available the way, unlock team would prefer it to be implemented that would be really helpful for the dev.

We can connect too, to discuss the design

**Comment by Ramitphi** [comment]
added designs for the same 

https://www.figma.com/file/wSEMAF2OZuTTl66wJM6Gnc/Unlock-Protocol---Huddle01-Bounty?type=design&node-id=1%3A215&mode=design&t=14pTTEGERVH6WWcg-1

@julien51 would love to know which flow you would prefer

**Comment by hangyakuzero** [comment]
Hey Julien, won't the address of the token be known after the lock is created, how do you suggest we implement a solution for this?

**Comment by julien51** [comment]
> Hey Julien, won't the address of the token be known after the lock is created, how do you suggest we implement a solution for this?

Yes it will... and I suggest you do that same as we do for our events page were basically we create the event object only once the tx has been executed.

**Comment by vrajdesai78** [comment]
Hey @julien51 what's the bounty prize for this issue

**Comment by julien51** [comment]
I like the first design more @Ramitphi ! 
When the box is checked, this should "disable" the `input` field!

We are attaching a 200 USDC bounty to that one @vrajdesai78 !

Please @vrajdesai78 or @hangyakuzero let me know if you are interested in working on this one and let's do it!

**Comment by vrajdesai78** [comment]
> I like the first design more @Ramitphi ! When the box is checked, this should "disable" the `input` field!
> 
> We are attaching a 200 USDC bounty to that one @vrajdesai78 !
> 
> Please @vrajdesai78 or @hangyakuzero let me know if you are interested in working on this one and let's do it!

Hey @julien51, I am happy to work on this issue.

**Comment by vrajdesai78** [comment]
@julien51 I need to call `create-room` API from server side only, so what is the best way to implement it. Will I need to create an API in locksmith and use it in `unlock-app`.

**Comment by julien51** [comment]
There is litterally a tutorial of how to build this with Huddle https://docs.huddle01.com/docs/guides/token-gated

I think creating a simple app that leverages Unlock + Huddle and makes it really easy for performers/creators to token gate their video streams could go very very far!

**Comment by bistenofficial** [comment]
Hi! I opened a PR for this bounty: https://github.com/unlock-protocol/unlock/pull/16394

It adds a server-side Locksmith endpoint for creating Huddle rooms, checks that the caller manages the deployed event lock, and wires the event form checkbox so the virtual event URL is replaced with the generated Huddle room after lock deployment.

**Comment by hangyakuzero** [comment]
I thought huddle pivoted and no longer has a meetings app

**Comment by wilber123451-design** [comment]
Hi @julien51, is the 200 USDC bounty still active? I verified the earlier scope and the three closed implementation attempts (#13687, #13689, #16394), but Huddle's current official docs now cover Cloud/AI infrastructure and the linked meetings/token-gated-room API is no longer available. If the bounty is still open, could you confirm whether the intended deliverable is (a) an updated integration with a replacement Huddle product/API, or (b) a standalone Unlock-gated video app using another maintained RTC provider? I can take it once the current scope and Base USDC payout are confirmed.
